### PR TITLE
[KYLIN-4938] remove segment by UUID

### PR DESCRIPTION
**Pull request by Ted-Jiang**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [ ] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java [hunk 1]
```diff
@@ -610,6 +610,32 @@ public CubeInstance rebuildLookupSnapshot(CubeInstance cube, String segmentName,
...
+        CubeSegment toDelete = null;
+
+        toDelete = cube.getSegmentById(uuid);
+
+        if (toDelete == null) {
+            throw new BadRequestException(String.format(Locale.ROOT, msg.getSEG_NOT_FOUND(), uuid));
+        }
+
+        if (cube.getStatus() == RealizationStatusEnum.DISABLED
+                || (cube.getStatus() == RealizationStatusEnum.READY && toDelete.getStatus() != SegmentStatusEnum.READY)) {
```
Comment (zhangayqian) [inline server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java:626]: When a segment with `NEW` status has related running jobs, I think it shouldn't be able to be deleted. Method `CubeService.isOrphonSegment(CubeInstance cube, String segId)` can help to judge whether a segment has related jobs.

File: server-base/src/main/java/org/apache/kylin/rest/msg/Message.java [hunk 2]
```diff
@@ -65,6 +65,10 @@ public String getDELETE_NOT_READY_SEG() {
         return "Cannot delete segment '%s' as its status is not READY. Discard the on-going job for it.";
     }
 
+    public String getDELETE_READY_SEG_BY_UUID() {
+        return "Cannot delete segment by UUID '%s' as its status is READY or its Cube is READY. ";
```
Comment (zhangayqian) [inline server-base/src/main/java/org/apache/kylin/rest/msg/Message.java:69]: Remove the space at the end to keep the format uniform.

**Review by zhangayqian — CHANGES_REQUESTED** [review]

File: server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java (hunk 1 above)
Comment (Ted-Jiang) [reply server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java]: good catch!  fix.

File: server-base/src/main/java/org/apache/kylin/rest/msg/Message.java (hunk 2 above)
Comment (Ted-Jiang) [reply server-base/src/main/java/org/apache/kylin/rest/msg/Message.java]: fix.

**Review by zhangayqian — APPROVED** [review]

**Comment by zhangayqian** [comment]
Thank you @Ted-Jiang
LGTM

**Review by hit-lacus — APPROVED** [review]
Approved

**Pull request merged**